fineract-provider/src/main/java/org/apache/fineract/infrastructure/hooks/processor/MessageGatewayHookProcessor.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Set;
+import org.apache.fineract.infrastructure.core.exception.GeneralPlatformDomainRuleException;
+import org.apache.fineract.infrastructure.hooks.domain.Hook;
+import org.apache.fineract.infrastructure.hooks.domain.HookConfiguration;
+import org.apache.fineract.infrastructure.sms.domain.SmsMessage;
+import org.apache.fineract.infrastructure.sms.domain.SmsMessageRepository;
+import org.apache.fineract.infrastructure.sms.scheduler.SmsMessageScheduledJobService;
+import org.apache.fineract.portfolio.client.domain.Client;
+import org.apache.fineract.portfolio.client.domain.ClientRepositoryWrapper;
+import org.apache.fineract.template.domain.Template;
+import org.apache.fineract.template.domain.TemplateRepository;
+import org.apache.fineract.template.service.TemplateMergeService;
+import org.apache.fineract.useradministration.domain.AppUser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MessageGatewayHookProcessor implements HookProcessor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MessageGatewayHookProcessor.class);
+
+    private final ClientRepositoryWrapper clientRepository;
+    private final TemplateRepository templateRepository;
+    private final TemplateMergeService templateMergeService;
+
+    private final SmsMessageRepository smsMessageRepository;
+    private final SmsMessageScheduledJobService smsMessageScheduledJobService;
+
+    @Autowired
+    public MessageGatewayHookProcessor(ClientRepositoryWrapper clientRepository, TemplateRepository templateRepository,
+            TemplateMergeService templateMergeService, SmsMessageRepository smsMessageRepository,
+            SmsMessageScheduledJobService smsMessageScheduledJobService) {
+        this.clientRepository = clientRepository;
+        this.templateRepository = templateRepository;
+        this.templateMergeService = templateMergeService;
+        this.smsMessageRepository = smsMessageRepository;
+        this.smsMessageScheduledJobService = smsMessageScheduledJobService;
+    }
+
+    @Override
+    public void process(final Hook hook, @SuppressWarnings("unused") final AppUser appUser, final String payload, final String entityName,
+            final String actionName, final String tenantIdentifier, final String authToken) throws IOException {
+
+        final Set<HookConfiguration> config = hook.getHookConfig();
+
+        Integer SMSProviderId = null;
+
+        for (final HookConfiguration conf : config) {
+            final String fieldName = conf.getFieldName();
+            if (fieldName.equals(SMSProviderIdParamName)) {
+                SMSProviderId = Integer.parseInt(conf.getFieldValue());
+            }
+        }
+
+        String templateName = entityName + "_" + actionName;
+
+        // 1 : find template via mapper using entity and action
+        Template template = null;
```

Comment:
```suggestion
        Template template;
```

is better, because it will be assign in the `if .. else` below anyway, so it's "safer" not to initialize it with null